src/overlays/actors/ovl_En_Bili/z_en_bili.c

```diff
@@ -46,69 +59,714 @@ static ColliderCylinderInit D_809C1640 = {
+    this->collider.base.atFlags &= ~AT_ON;
+    this->actionFunc = EnBili_Stunned;
+}
+
+void EnBili_SetupFrozen(EnBili* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f effectPos;
+
+    if (!(this->actor.flags & 0x8000)) {
+        this->actor.gravity = -1.0f;
+    }
+
+    this->actor.velocity.y = 0.0f;
+    effectPos.y = this->actor.world.pos.y - 15.0f;
+
+    for (i = 0; i < 8; i++) {
+
+        effectPos.x = this->actor.world.pos.x + ((i & 1) ? 7.0f : -7.0f);
+        effectPos.y += 2.5f;
+        effectPos.z = this->actor.world.pos.z + ((i & 4) ? 7.0f : -7.0f);
+
+        EffectSsEnIce_SpawnFlyingVec3f(globalCtx, &this->actor, &effectPos, 150, 150, 150, 250, 235, 245, 255,
+                                       (Rand_ZeroOne() * 0.2f) + 0.7f);
+    }
+
+    this->actor.speedXZ = 0.0f;
+    func_8003426C(&this->actor, 0, 0x96, 0x2000, 0xA);
+    this->collider.base.atFlags &= ~AT_ON;
+    this->collider.base.acFlags &= ~AC_ON;
+    this->timer = 300;
+    this->actionFunc = EnBili_Frozen;
+}
+
+// Miscellaneous
+
+/**
+ * Changes the texture displayed on the oral arms limb using the current frame.
+ */
+void EnBili_UpdateOralArmsIndex(EnBili* this) {
+    s16 curFrame = this->skelAnime.curFrame;
+    s16 temp; // Not strictly necessary, but avoids a few s16 casts
+
+    if (this->actionFunc == EnBili_DischargeLightning) {
+        temp = 3 - curFrame;
+        this->oralArmsTexIndex = (ABS(temp) + 5) % 8;
+    } else if (this->actionFunc == EnBili_Climb) {
+        if (curFrame < 10) {
+            temp = curFrame >> 1;
+            this->oralArmsTexIndex = CLAMP_MAX(temp, 3);
+        } else if (curFrame < 19) {
+            temp = 17 - curFrame;
+            this->oralArmsTexIndex = CLAMP_MIN(temp, 0) >> 1;
+        } else if (curFrame < 37) {
+            this->oralArmsTexIndex = ((36 - curFrame) / 3) + 2;
+        } else {
+            this->oralArmsTexIndex = (40 - curFrame) >> 1;
+        }
+    } else {
+        this->oralArmsTexIndex = curFrame >> 1;
+    }
+}
+
+/**
+ * Tracks Player height, with oscillation, and moves away from walls
+ */
+void EnBili_UpdateFloating(EnBili* this) {
+    f32 playerHeight = this->actor.world.pos.y + this->actor.yDistToPlayer;
+    f32 heightOffset = ((this->actionFunc == EnBili_SetNewHomeHeight) ? 100.0f : 40.0f);
+    f32 baseHeight = CLAMP_MIN(this->actor.floorHeight, playerHeight);
+
+    Math_StepToF(&this->actor.home.pos.y, baseHeight + heightOffset, 1.0f);
+    this->actor.world.pos.y = this->actor.home.pos.y + (sinf(this->timer * (M_PI * 0.0625f)) * 3.0f);
+
+    // Turn around if touching wall
+    if (this->actor.bgCheckFlags & 8) {
+        this->actor.world.rot.y = this->actor.wallYaw;
+    }
+}
+
+// Action functions
+
+void EnBili_FloatIdle(EnBili* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    if (!(this->timer % 4)) {
```

Comment:
```suggestion
    if ((this->timer % 4) == 0) {
```

Reply:
This doesn't work: the compiler simplifies it into `& 4 == 0`